### DISCUSSION TITLE
Re-Install behavior of pterodactyl

### DISCUSSION THREAD

**Discussion by fragger1991** [pterodactyl/panel#4055] [Support] [answered]
I noticed that the reinstall function works differently than I would expect.

When I start a reinstallation, the game server files that have already been downloaded are typically all retained. This means that if the server is defective (e.g. accidentally deleted too much), a reinstallation will not solve the problem.

I would expect the docker container to be completely wiped and rebuilt. But that doesn't seem to be the case.

Can someone help me and sharpen my understanding of the function a bit? In particular, I would be interested in what exactly the function does and why it only does it that way.

Thanks!

**Comment by parkervcp** [comment, marked as answer]
A reinstall re-runs the install script, and unless it's part of the install script it will keep all the files.

The server files are not kept in the container but in a mount that is re-mounted on start and stop.
